eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -127,7 +130,7 @@ jobs:
       artifact: ${{ parameters.reuseBuildArtifactsFrom }}_${{ parameters.architecture }}_Artifacts
       patterns: |
         **/Private.SourceBuilt.Artifacts.*.tar.gz
-        **/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])*.tar.gz
+        **/dotnet-sdk-*.tar.gz
```

Comment:
Why was this change necessary?

Reply:
This change was made in https://github.com/dotnet/installer/commit/f54f69f9aade0a00936aca343c3eb493d5edbc05. I must've missed reverting it back when I copied over my changes from 8.0.1xx to main.

Reply:
[Reverted](https://github.com/dotnet/installer/pull/17370/commits/18c19f97403225fa3f73c48582ba0805d9080387#diff-dc549eeb99aa73960e74c9d4c160a3c76e3e414398bb348c1d5365aac0369ad5R133)